redhat-developer/observability-operator
# Self reporting metrics

## Description
Jira: [MGDSTRM-9468](https://issues.redhat.com/browse/MGDSTRM-9468)

This PR adds reconciliation and configuration sync metrics to the Operator. Specifically the following metrics are added:
  * reconciler_total_count
  * reconciler_failure_count
  * configuration_syncs_total_count
  * configuration_sync_failure_count

## Verification
* Run this PR locally
* Use `curl 0.0.0.0:8080/metrics | grep observability_operator` to view metrics
* `reconciler_total_count` reports metrics by stage (Prometheus, Grafana, configuration etc.)
* A configuration failure can be triggered by altering the `kafka-observability-configuration` secret (for example removing the repository link). This failure should be counted in the  `configuration_sync_failure_count` metric.
